src/app/application.cpp

```diff
@@ -186,7 +186,21 @@ void Application::allTorrentsFinished()
         else if (shutdown)
             action = ShutdownAction::Shutdown;
 
-        if (!ShutdownConfirmDlg::askForConfirmation(action)) return;
+        if (will_exit_only) {
+            if (!pref->dontConfirmAutoExit()) {
+                bool exitConfirmed = false;
+                bool neverAskForExitConfirmationAgain = false;
+                ShutdownConfirmDlg::askForConfirmation(action, &exitConfirmed, &neverAskForExitConfirmationAgain);
```

Comment:
You don't need to modify the function singrature so much. Return a bool to signify a success/failure. Also take the bool by reference instead.
It should look like this:
```c++
// Function signature
bool ShutdownConfirmDlg::askForConfirmation(const ShutdownAction &action, bool &neverShowAgain = false);

// Calling it like this
if (!ShutdownConfirmDlg::askForConfirmation(action, neverAskForExitConfirmationAgain))
  return;

if (neverAskForExitConfirmationAgain)
    pref->setDontConfirmAutoExit(true);
```